nav2_smac_planner/src/a_star.cpp

```diff
@@ -234,11 +234,6 @@ bool AStarAlgorithm<NodeT>::areInputsValid()
     throw nav2_core::GoalOccupied("Goal was in lethal cost");
   }
 
-  // Check if starting point is valid
-  if (!_start->isNodeValid(_traverse_unknown, _collision_checker)) {
```

Comment:
Why not keep if it'll auto be OK?

Reply:
I would argue that it misleads the reader. But I can add it back.

Reply:
I can see that. How about putting a comment inline then saying we don't check due to the clearing under start condition? Something that makes me in 5 years not look at this and wonder why I skipped it while skimming

Reply:
Added in the comment. 

The error that would be throw in the case you are mentioning is `NoValidPathCouldBeFound`. 

So, 

You could say that if the expansions < number of neighbors(assuming no prunning) then the start is invalid.
 
If we expect that the robot could be a few cells deep of lethal cost then we probably should clear the whole footprint of the robot rather than just a cell.

Reply:
> You could say that if the expansions < number of neighbors(assuming no prunning) then the start is invalid.

Exactly. Though I think if you tested it, it might actually end up being just `0` or `1`, since the getNeighbors would return no possible motion, so we wouldn't be `iterations++`-ing. I'd test to verify, but I think it should be one of those two. If not, then yea, less than the number of neighbors is a reasonable choice